src/audio/recorder_wav.cpp

```
@@ -0,0 +1,163 @@
+}
+
+void Recorder::finalize_wav_header() {
+    if (!file_.is_open()) return;
+
+    int64_t total_samples = samples_written_.load();
+    // Compute data size in 64-bit to avoid overflow
+    int64_t data_size_64 = total_samples * static_cast<int64_t>(channels_) * 2;
+    // WAV format uses 32-bit sizes; clamp to max int32
+    int data_size = static_cast<int>(
+        (data_size_64 > 0x7FFFFFFF) ? 0x7FFFFFFF : data_size_64);
+    int riff_size = data_size + 36;
```

Cppcheck (2.20.0)

[error] Signed integer overflow for expression 'data_size+36'.

(integerOverflow)